utils/browser_history.jsx

```diff
@@ -42,4 +44,5 @@ export const browserHistory = {
             b.push(path, ...args);
         }
     },
-};
+});
+export const browserHistory = getModule('utils/browser_history');
```

Comment:
Another related question, how often would you expect that we're overriding non-component modules for the desktop app? I'm trying to think of how much getting a clean pattern for injecting this type of module should be since if we're only doing it in a handful of places, then getting a cleaner pattern is less necessary.

For injectible components themselves, the existing `Pluggable` system could likely be adapted to support that just fine since it already supports things like default components and plays well with React.

Reply:
We would have to defer to a dynamic import if we wanted to do it at the file level, meaning we'd likely be in the realm of top-level awaits to make sure that file is loaded, OR everything that imports it would need to understand that it's dynamic.

Currently browserHistory is the only one that's needed, but I'm thinking the websocket module would be another because currently I'm overriding the native JS WebSocket object to make sure the WS URL is correct. Since that's not very clean, I think overriding the module we use to do that work would make more sense.

Reply:
Right. Any sort of dynamic import that happens at runtime would be really tricky. We could theoretically make a babel plugin or something to make that happen automatically, but that seems like overkill.

What if instead of thinking about replacing the whole file (which might have just been me), we instead change it so we do the singleton thing a bit better by having a function `getHistory` to return the instance of the singleton rather than just exporting `browserHistory` directly. I'm having flashbacks to how it was better in Java to have a singleton accessed through a static method like `History.getInstance()` rather than just exposing `History.INSTANCE` as a static variable. It'll obviously require some changes where we use the history object, but if we do the singleton better, we can easily access the module registry or do whatever we want without having to do any magic.

I think something like
```js
const history = {/* whatever the web app does now*/};

export function getHistory() {
    return getModule('utils/browser_history') ?? history;
}
```

That would also require less changes if we were to use the `useHistory` hook from react-router-dom in components since we shouldn't need to access `utils/browser_history` as often as we do.